Xamarin.Forms.Platform.Android/Renderers/NavigationRenderer.cs

```diff
@@ -145,7 +145,7 @@ protected virtual Task<bool> OnPushAsync(Page view, bool animated)
 		void InsertPageBefore(Page page, Page before)
 		{
 
-			int index = ((IPageController)Element).InternalChildren.IndexOf(before);
+			int index = PageController.InternalChildren.IndexOf(before);
```

Comment:
this code was already in.. and isn't related..  and here Element with 99% will not be null